### PR TITLE
Use builtin ISO date functions

### PR DESCRIPTION
**Pull request by Joe-Heffer-Shef**
Use standard Python functions

### CODE REVIEW

File: environment_agency_flood/env_agency_flood.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def get_data(self) -> iter:
             endpoint = 'id/stations/{station_id}/readings.csv'.format(station_id=self.get_value(station,"items_stationReference"))
             
             params = dict(
-                date=self.date.strftime("%Y-%m-%d"),
```
Comment (vekimenko) [inline environment_agency_flood/env_agency_flood.py]: isoformat() is not working for me. It's probably versions collision. I am using Python 3.5.

File: environment_agency_flood/env_agency_flood.py [hunk 2]
```diff
@@ -360,7 +360,7 @@ def get_args() -> argparse.Namespace:
     """Command-line arguments"""
 
     parser = argparse.ArgumentParser(description=DESCRIPTION)
-    parser.add_argument('-d', '--date', required=True, type=lambda s: datetime.datetime.strptime(s, '%Y-%m-%d'), help="ISO UTC date")
+    parser.add_argument('-d', '--date', required=True, type=datetime.date.fromisoformat, help="ISO UTC date")
```
Comment (vekimenko) [inline environment_agency_flood/env_agency_flood.py]: Python 3.5 datetime doesn't have date. Which version is on the server?

Comment (Joe-Heffer-Shef) [reply environment_agency_flood/env_agency_flood.py]: I don't know how Python is set up on the server. There's no reason we can't use virtual environments for each pipeline. This means we should be able to use a recent Python version such as 3.6 or newer.

File: environment_agency_flood/env_agency_flood.py (hunk 1 above)
Comment (Joe-Heffer-Shef) [reply environment_agency_flood/env_agency_flood.py]: It should be in version 3.5
https://docs.python.org/3.5/library/datetime.html#datetime.date.isoformat

File: environment_agency_flood/env_agency_flood.py (hunk 2 above)
Comment (Joe-Heffer-Shef) [reply environment_agency_flood/env_agency_flood.py]: The basic installed version on the operating system is 2.7 and 3.5. I don't know whether or not a virtual environment is used to run the actual pipelines.

```
sa_cs1jsth@ufacqt:~$ python --version
Python 2.7.12

sa_cs1jsth@ufacqt:~$ python3 --version
Python 3.5.2
```

**Comment by Joe-Heffer-Shef** [comment]
I might cancel this PR and use Python 3.5 in my development environment

**Comment by Joe-Heffer-Shef** [comment]
sticking to Python 3.5

**Pull request closed without merging**